### DISCUSSION TITLE
download m3u instead of opening in browser?

### DISCUSSION THREAD

**Discussion by aniel300** [euzu/tuliprox#131] [General]
is it possible so when i go the m3u url like `https://m3ufilter.xxx.top/get.php?username=external&password=external&type=m3u_plus` or `https://m3ufilter.xxx.top/get.php?username=external&password=external` to download instead of openin in the browser?

**Comment by aniel300** [comment]
when i tried to download with idm i get this however i think is not downloading the whole playlist as it is missing stuff (so far i have notice it missing all the series)
![image](https://github.com/user-attachments/assets/d6a9cec7-42ef-4db6-b045-46957db1edcc)

**Reply by euzu** [reply]
what is idm ?

**Reply by aniel300** [reply]
https://www.internetdownloadmanager.com/

**Reply by euzu** [reply]
Opening a browser has to do with idm  and not m3u-filter.

**Reply by aniel300** [reply]
yes i know. the idm thing i just discovered. right now am just trying to pass trough everything but it seems when i use m3u in the target it is missing the tv shows. however, if i use xtream then it works but am also having another issue with that.

**Comment by euzu** [comment]
i dont understand the question,  What do you want to download ? the content of the m3u playlist ?  Then you need to use `wget` or `curl` like application, Btw `type=m3u_plus`  is ignored by m3u-filter. 
And if you use a iptv client it will doenload the content.  What exactly are you trying ?

**Reply by aniel300** [reply]
at this point am trying to filter and pass all he merged providers, etc over to another proxy, at least until and if u can implement some of the features i need

**Reply by euzu** [reply]
First:  When you have xtream input  and m3u output you loose the series. Why? because xtream api delivers you an overview for series, not the episodes, You need to fetch the episodes for every single series. This is possible by some options `xtream_resolve_series, xtream_resolve_series_delay` in m3u-filter, but the drawback is that your provider will eventually  ban you because of flooding.

**Reply by euzu** [reply]
What features are you missing ?

**Reply by aniel300** [reply]
> What features are you missing ?

https://github.com/euzu/m3u-filter/issues/117 and https://github.com/euzu/m3u-filter/issues/116

**Reply by aniel300** [reply]
> First: When you have xtream input and m3u output you loose the series. Why? because xtream api delivers you an overview for series, not the episodes, You need to fetch the episodes for every single series. This is possible by some options `xtream_resolve_series, xtream_resolve_series_delay` in m3u-filter, but the drawback is that your provider will eventually ban you because of flooding.

umm interesting. let me see if i can fix the xtream target not working and we will go from there. did u got a chance to get it working in ur end? https://github.com/euzu/m3u-filter/discussions/123

**Reply by euzu** [reply]
This features are not simple to implement thats why  need to ask you:

- buffering is done by the clients, it makes no difference to  a client if it buffers it self or m3u.filter buffers. If the provider does not stream fast enough or has hickups, buffering on both ends will change nothing.  If you know it could be please than explain to me how.
- Sharing the same  stream between clients is only possible  for live tv, vod and series are to indiviual to share. And  it benefits you only if you have a lot of users. When you have only a handfull how possible is it that they watch the same channel at the same time ? Explain me the real benefit of this features.

**Reply by euzu** [reply]
> > First: When you have xtream input and m3u output you loose the series. Why? because xtream api delivers you an overview for series, not the episodes, You need to fetch the episodes for every single series. This is possible by some options `xtream_resolve_series, xtream_resolve_series_delay` in m3u-filter, but the drawback is that your provider will eventually ban you because of flooding.
> 
> umm interesting. let me see if i can fix the xtream target not working and we will go from there. did u got a chance to get it working in ur end? #123

Is it not working with the Windows IPTVPlayer? i cant remember have to check

**Reply by aniel300** [reply]
1) the retry option is to make the streams stable. as in if it gets disconnected from the provider, etc, the proxy should restart the connection  etc in the background, but the client shouldn't notice what is happening in the background aka it just keeps streaming without cutting off, etc.

2) yes, am fine with it only being for live tv. i dont have that many users but i also have limited cxn to the providers. so, the reason behind this feature is to save on upload bw for the server and also not requiring extra cxn from the provider when 2 or more users are watching the same channel.

**Reply by aniel300** [reply]
> > > First: When you have xtream input and m3u output you loose the series. Why? because xtream api delivers you an overview for series, not the episodes, You need to fetch the episodes for every single series. This is possible by some options `xtream_resolve_series, xtream_resolve_series_delay` in m3u-filter, but the drawback is that your provider will eventually ban you because of flooding.
> > 
> > 
> > umm interesting. let me see if i can fix the xtream target not working and we will go from there. did u got a chance to get it working in ur end? #123
> 
> Is it not working with the Windows IPTVPlayer? i cant remember have to check

Windows IPTVPlayer give some errors/i did managed to load groups but same issue as sfvip and sfvip load the groups but cant play anything

**Reply by aniel300** [reply]
closing for the time being

**Reply by aniel300** [reply]
> 1. the retry option is to make the streams stable. as in if it gets disconnected from the provider, etc, the proxy should restart the connection  etc in the background, but the client shouldn't notice what is happening in the background aka it just keeps streaming without cutting off, etc.
> 2. yes, am fine with it only being for live tv. i dont have that many users but i also have limited cxn to the providers. so, the reason behind this feature is to save on upload bw for the server and also not requiring extra cxn from the provider when 2 or more users are watching the same channel.

did u give me an answer to this? @euzu

**Reply by euzu** [reply]
I am working on 2. its wip.

**Reply by aniel300** [reply]
ok i can create a testing container to test if needed. if u can implement 1 and 2. i can then not have to use two different projects. thank u for the effort

**Reply by euzu** [reply]
can you give me an example of 1. Which project implemented it and is it working ?

**Reply by aniel300** [reply]
this project did. i think i was the one who initially asked for this feature or someone else did. it we arrive to this feature by trial and error. it is working and it does make stream more stable with the occasional hiccups am assuming because there still some logic to be implemented to fully fix this behavior. https://github.com/sonroyaalmerol/m3u-stream-merger-proxy. an example would be provider disconnecting ur cxn from their end for x reason or aka bad providers that configured their panel in such a way that makes them a bad provider.

**Comment by aniel300** [comment]
closing this for now as i might or might not have the need for the m3u to download as opposed to open in the browser as it does now.
